geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -714,12 +734,20 @@ object DeltaWriter extends StrictLogging {
     val mappingsMap = mappingsBuilder.result()
 
     logger.trace(s"batch dictionary mappings: ${mappingsMap.mapValues(_.mkString(",")).mkString(";")}")
-    MergedDictionaries(dictionaryMap, mappingsMap)
+    MergedDictionaries(dictionaryMap, mappingsMap, allocator)
   }
 
   // holder for merged dictionaries and mappings from written values to merged values
-  private case class MergedDictionaries(dictionaries: Map[String, ArrowDictionary],
-                                        mappings: Map[String, Array[scala.collection.Map[Integer, Integer]]])
+  private case class MergedDictionaries(
+      dictionaries: Map[String, ArrowDictionary],
+      mappings: Map[String, Array[scala.collection.Map[Integer, Integer]]],
+      allocator: BufferAllocator
```

Comment:
Is the allocator passed in here so that it can be closed at the end?  

E.g. is this just here for 'clean up' later in the lifecycle?

Reply:
right